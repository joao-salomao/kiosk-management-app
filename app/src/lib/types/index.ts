export type Kiosk = {
  id: string;
  serialKey: string;
  description: string;
  isKioskClosed: boolean;
  storeOpensAt: string;
  storeClosesAt: string;
};

export enum StatusFilterEnum {
  open = "open",
  closed = "closed",
  all = "all",
}
