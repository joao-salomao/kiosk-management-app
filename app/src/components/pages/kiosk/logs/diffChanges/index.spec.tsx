import { Log } from 'lib/types';
import renderer from 'react-test-renderer';
import { DiffChanges } from ".";

const log = {
  description: JSON.stringify({
    from: {
      name: "Old name",
      age: 30,
      likesBeer: false,
    },
    to: {
      name: "New name",
      age: 31,
      likesBeer: true,
    }
  })
}

it('renders correctly', () => {
  const tree = renderer
    .create(
      <DiffChanges log={log as Log} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});