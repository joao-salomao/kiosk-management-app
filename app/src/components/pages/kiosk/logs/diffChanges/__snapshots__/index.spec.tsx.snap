// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p
    className="font-bold"
  >
    Changes:
  </p>
  <div>
    <span
      className="underline"
    >
      #name
    </span>
     from "
    Old name
    " to "
    New name
    "
  </div>
  <div>
    <span
      className="underline"
    >
      #age
    </span>
     from "
    30
    " to "
    31
    "
  </div>
  <div>
    <span
      className="underline"
    >
      #likesBeer
    </span>
     from "
    No
    " to "
    Yes
    "
  </div>
</div>
`;
